### PR TITLE
WT-11353 Update the "Writing WiredTiger applications" section with background compaction

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: src/docs/compact.dox [hunk 1]
```diff
@@ -2,11 +2,49 @@
...
+# Execute compaction on "mytable" with the default parameters.
+int ret = session->compact(session, "table:mytable", NULL);
+
+# Execute compaction on "mytable" if at least 20MB of free space can be theoretically reclaimed by compaction.
+int ret = session->compact(session, "table:mytable", "free_space_target=20MB");
+@endcode
+
+@section background Background compaction
+
+It is possible to interact with background compaction through the configuration item "background".
```
Comment (kommiharibabu) [inline src/docs/compact.dox:25]: This sentence feels like it is not completed. How about adding "background is provided to the WT_SESSION::compact method"?

File: src/docs/compact.dox [hunk 2]
```diff
@@ -2,11 +2,49 @@
...
+The call immediately returns to the application but does not guarantee compaction will be enabled/disabled at the same time.
+A successful call simply means the request has been submitted to WiredTiger.
+
+When background compaction is enabled, WiredTiger autonomously compacts the eligible files in the database.
+
+<b>Note:</b> When interacting with background compaction, NULL should be assigned to the argument corresponding to the table name.
```
Comment (kommiharibabu) [inline src/docs/compact.dox:31]: `table` to `uri`? API specifies `uri`?

File: src/docs/compact.dox [hunk 3]
```diff
@@ -2,11 +2,49 @@
...
 
-Because checkpoints named by the application are not discarded until
-explicitly removed or replaced, they may prevent WT_SESSION::compact
-from accomplishing anything.
+<b>Note:</b> Because checkpoints named by the application are not discarded until explicitly removed or replaced, they may prevent WT_SESSION::compact from accomplishing anything.
+
+@section foreground Foreground compaction
+
+Foreground compaction is executed when a specific table is given to the WT_SESSION::compact method.
+The call returns once compaction has been executed on the table.
```
Comment (kommiharibabu) [inline src/docs/compact.dox:12]: `table` to `specified uri`?

File: src/docs/compact.dox (hunk 1 above)
Comment (etienneptl) [reply src/docs/compact.dox]: Thank you, I reworded in 2d3836de9cf37abd9ab8e6f9cc092568b0d67ca1, let me know what you think.

File: src/docs/compact.dox (hunk 2 above)
Comment (etienneptl) [reply src/docs/compact.dox]: Used URI in 47ae7c97c2ebca42ee19132458818f0ca1ade75f

File: src/docs/compact.dox (hunk 3 above)
Comment (etienneptl) [reply src/docs/compact.dox]: Used URI in 47ae7c97c2ebca42ee19132458818f0ca1ade75f

File: src/docs/compact.dox [hunk 4]
```diff
@@ -2,11 +2,50 @@
...
+
+# Execute compaction on "mytable" if at least 20MB of free space can be theoretically reclaimed by compaction.
+int ret = session->compact(session, "table:mytable", "free_space_target=20MB");
+@endcode
+
+@section background Background compaction
+
+The "background" configuration item can be used to interact with background compaction when calling
+the WT_SESSION::compact method. The call immediately returns to the application but does not
+guarantee compaction will be enabled/disabled at the same time. A successful call simply means the
```
Comment (jiechenbo) [inline src/docs/compact.dox:27]: Could you elaborate on this part? 
```
A successful call simply means the request has been submitted to WiredTiger.
```

File: src/docs/compact.dox [hunk 5]
```diff
@@ -2,11 +2,50 @@
...
+request has been submitted to WiredTiger.
+
+When background compaction is enabled, WiredTiger autonomously compacts the eligible files in the database.
```
Comment (jiechenbo) [inline src/docs/compact.dox:30]: Do we document on what an eligible file is here?

File: src/docs/compact.dox [hunk 6]
```diff
@@ -2,11 +2,50 @@
 
 The WT_SESSION::compact method can be used to compact data sources.
 
-The data source does not have to be quiescent, compaction may be
-performed on an active data source.
+The data source does not have to be quiescent, compaction may be performed on an active data source.
```
Comment (jiechenbo) [inline src/docs/compact.dox:5]: What does this mean here? Maybe it's worth a rewrite

File: src/docs/compact.dox (hunk 4 above)
Comment (etienneptl) [reply src/docs/compact.dox:27]: Sure, see f0952709581d991ca1b5ccf6aad511487598677e

File: src/docs/compact.dox (hunk 5 above)
Comment (etienneptl) [reply src/docs/compact.dox:31]: We do in the arch guide, added a reference to it in 36c8c4f5c6ee094e14c1a238dc545425ebdfac18

File: src/docs/compact.dox (hunk 6 above)
Comment (etienneptl) [reply src/docs/compact.dox]: I rewrote that section here 537812b6b3d07314f48896adc7cab31be3390a38, let me know what you think.

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Pull request merged**